pyaedt/application/Design.py

```diff
@@ -4029,33 +4032,65 @@ def set_temporary_directory(self, temp_dir_path):
 
-        References
-        ----------
+    References
+    ----------
 
-        >>> oDesign.GetChildObject("Design Settings")
-        """
+    >>> oDesign.GetChildObject("Design Settings")
+    """
+
+    def __init__(self, app):
+        self._app = app
+        self.manipulate_inputs = None
         try:
-            design_settings = self._odesign.GetChildObject("Design Settings")
-        except Exception:  # pragma: no cover
-            self.logger.error("Failed to retrieve design settings.")
-            return False
+            self.design_settings = self._app.odesign.GetChildObject("Design Settings")
+        except GrpcApiError:  # pragma: no cover
+            self._app.logger.error("Failed to retrieve design settings.")
+            self.design_settings = None
+
+    @property
+    def available_properties(self):
+        """Available properties names for the current design."""
+        return [prop for prop in self.design_settings.GetPropNames() if not prop.endswith("/Choices")]
+
+    def __repr__(self):
+        lines = ["{"]
+        for prop in self.available_properties:
+            lines.append("\t{}: {}".format(prop, self.design_settings.GetPropValue(prop)))
+        lines.append("}")
+        return "\n".join(lines)
+
+    def __setitem__(self, key, value):
+        if key in self.available_properties:
+            if self.manipulate_inputs is not None:
+                value = self.manipulate_inputs.execute(key, value)
+            key_choices = "{}/Choices".format(key)
+            if key_choices in self.design_settings.GetPropNames():
+                value_choices = self.design_settings.GetPropValue(key_choices)
+                if value not in value_choices:
+                    self._app.logger.error(
+                        "{} is not a valid choice. Possible choices are: {}".format(value, ", ".join(value_choices))
+                    )
+                    return False
+            self.design_settings.SetPropValue(key, value)
+        else:
+            self._app.logger.error("{} property is not available in design settings.".format(key))
+
+    def __getitem__(self, key):
+        if key in self.available_properties:
+            return self.design_settings.GetPropValue(key)
+        else:
+            self._app.logger.error("{} property is not available in design settings.".format(key))
+            return None
+
+    def __contains__(self, item):
+        return item in self.available_properties
 
-        prop_name_list = design_settings.GetPropNames()
-        design_settings_dict = {}
-        for prop in prop_name_list:
-            try:
-                design_settings_dict[prop] = design_settings.GetPropValue(prop)
-            except Exception:  # pragma: no cover
-                self.logger.warning('Could not retrieve "{}" property value in design settings.'.format(prop))
-                design_settings_dict[prop] = None
 
-        return design_settings_dict
+class DesignSettingsManipulation:
```

Comment:
> Would it be possible to define a DesignSettingsManipulation interface which IcepakDesignSettingsManipulation should implement ? The idea would be to "follow" the strategy design pattern. However, I don't know to what extent this is compatible with IronPython (if these line of codes are concerned ofc)

@SMoraisAnsys Added an abstract class here

Reply:
Awesome, glad to know this could be compatible with IronPython !